src/zarr/core/metadata/v2.py

```diff
@@ -165,18 +157,44 @@ def update_attributes(self, attributes: dict[str, JSON]) -> Self:
         return replace(self, attributes=attributes)
 
 
-def parse_zarr_format(data: Literal[2]) -> Literal[2]:
+def parse_zarr_format(data: object) -> Literal[2]:
```

Comment:
I think this should stay as `Literal[2]`? Even though it's possible to input any type here, giving it a narrow type allow errors to be spotted at type checking time, ie before runtime.

Reply:
`parse_zarr_format` is designed for use with unknown data. If we annotate the input type as `Literal[2]`, then mypy will complain if you try to call this function on unknown data, and we have defeated the point of the function. 

If you already know that a variable has type `Literal[2]`, then there is no need to call `parse_zarr_format`. I do think we should find a way to construct metadata documents from pre-validated data, but that's out of scope for this function.